### ISSUE TITLE
MySQL standard TCP/IP over SSH connection (namecheap.com MySQL)

### ISSUE THREAD

**Issue by EvanVS** [grafana/grafana#31840]
I just bought a domain and hosting from namecheap.com. They keep the MySQL port blocked on the firewall so you have to access the MySQL server over an SSH tunnel. I believe they did this to add more security as MySQL can be used to store login/user information and this would help prevent database attacks/leaks. It really sucks because one of the main things I wanted to do was access the MySQL data from Grafana.

My question, is there any current way to do this with Grafana that I missed or do I need to make this a feature request?

I don't have that much experience with SSH but I know what it does, I work more with the hardware side of things. Grafana is running on my home server running Windows Server 2019. I really tried to find a previous issue/feature request but couldn't find anything. Sorry if this is a duplicate or if I missed something that's already apart of Grafana.

**Comment by ying-jeanne** [comment]
Hi @EvanVS, I found this old ticket https://github.com/grafana/grafana/issues/24044, if your server support SSL CA, you could use https://grafana.com/docs/grafana/latest/administration/configuration/#ssl_mode to establish the connection.

hope it helps.

**Comment by Elfo404** [comment]
closing this as a duplicate of  #24044, as stated there, implementing ssh tunneling directly in grafana is unlikely to happen.

You may find some useful information on google on how to set-up a local ssh tunnel on windows.
